projects/igniteui-angular/src/lib/directives/text-selection/text-selection.directive.ts

```diff
@@ -17,17 +47,62 @@ export class IgxTextSelectionDirective {
         this.selectionState = val;
     }
 
+    /**
+     * Returns the nativeElement of the element where the directive was applied.
+     *
+     * ```html
+     * <input
+     *   type="text"
+    *    id="firstName"
+     *   igxTextSelection>
+     * </input>
+     * ```
+     *
+     * ```typescript
+     * @ViewChild('firstName',
+     *  {read: IgxTextSelectionDirective})
+     * public inputElement: IgxTextSelectionDirective;
+     *
+     * public getNativeElement() {
+     *  return this.inputElement.nativeElement;
+     * }
+     * ```
+     */
     get nativeElement() {
         return this.element.nativeElement;
     }
 
+    /**
+     * @hidden
+     */
     @HostListener('focus')
     onFocus() {
         this.trigger();
     }
 
     constructor(private element: ElementRef) { }
 
+    /**
+     * Triggers the selection of the element.
```

Comment:
Another proposa, it may sounds more accurate: Triggers the selection of the element if it is marked as though.